lib/msf/core/exploit/remote/ldap/server.rb

```diff
@@ -0,0 +1,127 @@
+# -*- coding: binary -*-
+
+module Msf
+
+###
+#
+# This module exposes methods for querying a remote LDAP service
+#
+###
+module Exploit::Remote::LDAP
+module Server
+  include Exploit::Remote::SocketServer
+
+  #
+  # Initializes an exploit module that serves LDAP requests
+  #
+  def initialize(info = {})
+    super
+
+    register_options(
+      [
+        OptPort.new('SRVPORT', [true, 'The local port to listen on.', 389]),
+        OptPath.new('LDIF_FILE', [ false, "Directory LDIF file path"]),
+      ], Exploit::Remote::LDAP::Server
+    )
+
+    register_advanced_options(
+      [
+        OptBool.new('LdapServerUdp', [true, "Serve UDP LDAP requests", true]),
+        OptBool.new('LdapServerTcp', [true, "Serve TCP LDAP requests", true])
+      ], Exploit::Remote::LDAP::Server
+    )
+  end
+
+  attr_accessor :service # :nodoc:
+
+  #
+  # Read LDIF file - from net-ldap ldapserver.rb#L162
+  #
+  # @ return [Hash] parsed ldif file
+  def read_ldif
+    if File.exists?(datastore['LDIF_FILE'])
```

Comment:
I was getting an exception here when I had not set the LDIF_FILE option. This was how I fixed it.
```suggestion
    if !datastore['LDIF_FILE'].blank? && File.exists?(datastore['LDIF_FILE'])
```

Reply:
Thanks, adding (i didn't hit that but my `OptX` code is different)